univocity-trader-core/src/main/java/com/univocity/trader/indicators/UlcerIndexIndicator.java

```diff
@@ -0,0 +1,70 @@
+package com.univocity.trader.indicators;
+
+import com.univocity.trader.candles.Candle;
+import com.univocity.trader.indicators.base.MultiValueIndicator;
+import com.univocity.trader.indicators.base.TimeInterval;
+import com.univocity.trader.strategy.Indicator;
+
+import java.util.function.ToDoubleFunction;
+
+import static com.univocity.trader.indicators.base.TimeInterval.MINUTE;
+
+public class UlcerIndexIndicator extends MultiValueIndicator {
+
+    private double value;
+
+    private HighestValueIndicator highestValueInd;
+    private double length;
+
+    public UlcerIndexIndicator(TimeInterval interval) {
+        this(14, interval);
+    }
+
+    public UlcerIndexIndicator(int length, TimeInterval interval) {
+        this(length, interval, c -> c.close);
+    }
+
+    public UlcerIndexIndicator(int length, TimeInterval interval, ToDoubleFunction<Candle> valueGetter) {
+        super(length, interval, valueGetter);
+        this.length = length;
+        highestValueInd = new HighestValueIndicator(length, interval, c -> c.close);
+    }
+
+    @Override
+    protected boolean calculateIndicatorValue(Candle candle, double value, boolean updating) {
+
+        final int index = Double.valueOf(candle.openTime / MINUTE.ms).intValue();
```

Comment:
This index calculation is incorrect. TA4J uses that to query its own candle history but here the calculation is always over relative to the last candle. If you need previous candles, the internal circular list has all the 'n' last candles available (where n = the length provided in the constructor). Also don't assume all candles are split by well-defined minute intervals - they can have 1ms, 40s, 1.324 hours, etc. 

I'll adjust that myself so you can check how to make it work with other indicators that also need `n` candles back from the current one.